meson.build

```
@@ -91,8 +91,10 @@ endif
 
 if meson.version().version_compare('>= 0.60')
   iconv_dep = dependency('iconv', required: get_option('iconv'))
+elif not cc.compiles('#include <iconv.h>\nint main{iconv_open("", "");}')
```

```suggestion
elif not cc.links('#include <iconv.h>\nint main{iconv_open("", "");}')
```

Use cc.links() for a full link test. It's not sufficient to check if it compiles, because that only proves that the header exists, which it will even if the cc.find_library() is necessary.

---

```suggestion
elif not get_option('iconv').disabled() and not cc.links('#include <iconv.h>\nint main{iconv_open("", "");}')
```

Checking for cc.links() here if the feature is disabled is a waste of a configure check.